catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardImpl.java

```diff
@@ -157,6 +157,9 @@ public MetacardImpl(Metacard metacard, MetacardType type) {
             throw new IllegalArgumentException(
                     MetacardType.class.getName() + " instance should not be null.");
         }
+        if (metacard.getSourceId() != null) {
```

Comment:
How confident are we that a null check is sufficient here, vs something like `StringUtils.isEmpty()`? If we *want* to allow zero-length/whitespace source ids, can we at least explain why?